### PR TITLE
feat(evmutil): implement MsgConvertCosmosCoinFromERC20

### PR DESCRIPTION
**Pull request by pirtleshell**
## Description
Adds message server implementation for MsgConvertCosmosCoinToERC20, with tests & spec updates.

## Checklist
 - [x] Changelog has been updated as necessary.

### CODE REVIEW

**Review by nddeluca — APPROVED** [review]
👍

File: x/evmutil/spec/01_concepts.md [hunk 1]
```diff
@@ -54,6 +54,8 @@ Only Cosmos co-chain denominations that are in the `AllowedCosmosDenoms` param (
 
 The ERC20 contracts are deployed and managed by x/evmutil. The contract is deployed on first convert of the coin. Once deployed, the addresses of the contracts can be queried via the `DeployedCosmosCoinContracts` query (`deployed_cosmos_coin_contracts` endpoint).
 
+If a denom is removed from the `AllowedCosmosDenoms` param, existing ERC20 tokens can be converted back to the underlying sdk.Coin via `MsgConvertCosmosCoinFromERC20`, but no conversions from sdk.Coin -> ERC via `MsgConvertCosmosCoinToERC20` are allowed.
```
[the remark this answers is not in the document]
Comment (pirtleshell) [reply x/evmutil/spec/01_concepts.md:57]: good idea, i've added a test of the full flow, including re-enabling the coin using the original contract address:
https://github.com/Kava-Labs/kava/pull/1609/files#diff-b41fd85eddf7b0e552e3551ed6991584ba52d5d9e7605f76facbb676139dadf1R654-R659

File: x/evmutil/keeper/msg_server_test.go [hunk 2]
```diff
@@ -509,3 +509,222 @@ func (suite *MsgServerSuite) TestConvertCosmosCoinToERC20_AlreadyDeployedContrac
...
+		})
+	}
+}
+
+// the test verifies the behavior for when a denom is removed from the params list
+// after conversions have been made:
+// - it should prevent more conversions from sdk -> evm for that denom
+// - existing erc20s should be allowed to get converted back to a sdk.Coins
+// - allowing the denom again should use existing contract
+func (suite *MsgServerSuite) TestConvertCosmosCoinForRemovedDenom() {
```
Comment (nddeluca) [inline x/evmutil/keeper/msg_server_test.go:659]: Great addition here 👍

Comment (nddeluca) [reply x/evmutil/keeper/msg_server_test.go:659]: I missed that last case of re-enabled when reviewing

**Pull request merged**